contrib/vdsm-dev

```diff
@@ -0,0 +1,187 @@
+    echo "Usage: $0 CMD [options]"
+    echo
+    echo "Available commands:"
+    echo "help:          Print this message and exits."
+    echo "build:         Build vdsm and make RPMs."
+    echo "build-install: Build vdsm, make RPMs, and update local packages."
+    echo "test:          Run any test from specified environment using tox."
+    echo "test-storage:  Run one/all storage tests using tox."
+    echo "show-coverage: Show coverage results generated after a test run."
+    echo "               Receives an environment with '-e env'. [default=storage]"
+}
+
+venv_create()
+{
+    echo "Creating venv at $venv_path"
+    python3 -m venv ${venv_path}
+    source "${venv_activate_path}"
+    pip install --upgrade pip
+    pip install -r docker/requirements.txt
+    deactivate
+}
+
+make_storage()
+{
+    if [ ! -d $user_storage_path ]; then
+        echo "Make storage at $user_storage_path"
+        make storage
+    fi
+}
+
+test_path()
+{
+    path=$1
+    storage_prefix=storage/
+    if [ ! -f "tests/$path" ]; then
+        path=${storage_prefix}${path}
+        if [ ! -f "tests/$path" ]; then
+            echo "No test $path found"
+            exit 1
+        fi
+    fi
+}
+
+vdsm_build()
+{
+    git clean -dxf && ./autogen.sh --system --enable-timestamp && \
+         make && rm -rf ~/rpmbuild && make rpm
+}
+
+handle_test_opts()
+{
+    while test $# -gt 0
+    do
+        case "$1" in
+            -e)
+                shift
+                env=$1
+                if [ $env == "storage" ]; then
+                    make_storage
+                fi
+                ;;
+            *)
+                # Let tox handle the rest of the arguments
+                ;;
+        esac
+        shift
+    done
+}
+
+
+if [ -z "$1" ]; then
+    echo "Command missing"
+    print_help
+    exit 1
+fi
+
+case $1 in
+
+    help )
+
+        print_help
+
+        ;;
+
+    build )
+
+        vdsm_build
+
+        ;;
+
+    build-install )
+
+        vdsm_build && (cd ~/rpmbuild/RPMS && sudo dnf upgrade */vdsm*.rpm)
+
+        ;;
+
+    test )
+
+        if [ ! -f $venv_activate_path ]; then
+            venv_create
+        fi
+
+        handle_test_opts "$@"
+
+        source ${venv_activate_path}
+        echo "Running: tox ${*:2}"
+        tox "${*:2}"
+
+        ;;
+
+    test-storage )
+
+        shift
+        test=$1
+
+        if [[ "$test" == *":"* ]]; then
+            IFS=':' read -a filepath rest <<<"$test"
+            test_path $filepath
+        else
+            test_path $test
+        fi
+        
+        if [ ! -f $venv_activate_path ]; then
+            venv_create
+        fi
+
+        make_storage
+
+        source ${venv_activate_path}
+        if [[ $test == "" ]]; then
+            echo "Running: tox -e storage"
+            tox -e storage
+        else
+            echo "Running: tox -e storage -- -vv $test ${*:2}"
+            tox -e storage -- -vv $test "${*:2}"
+        fi
```

Comment:
The main point is not avoiding to write this. The script handles the creation of the storage, creating (if required) and activating the venv, and then running the test. In that order. I still use it since I have it in my bin folder, and it is nice not having to worry about the venv activation/deactivation at all.

And if running from a new env (say, a new Vagrant VM), running a test through the script will generate the venv and the storage for you.
I think it can be improved to check also if it needs to run the `configure` and the first `make`.